python/lsst/source/injection/utils/_make_injection_pipeline.py

```diff
@@ -369,14 +388,17 @@ def make_injection_pipeline(
             pipeline.mergePipeline(additional_pipeline)
 
         # Add all tasks to subset_name. If the subset does not exist create it.
-        if isinstance(subset_name, str):
-            if subset_name in pipeline.subsets.keys():
-                for additional_task in additional_tasks:
-                    pipeline.addLabelToSubset(subset_name, additional_task)
-                    subset_grammar = f"the existing subset {subset_name}"
-            else:
-                pipeline.addLabeledSubset(subset_name, subset_description, additional_tasks)
-                subset_grammar = f"a new subset {subset_name}"
+        if isinstance(subset_names, str):
+            subset_names = subset_names.split(",")
```

Comment:
It looks like there are two different ways to supply multiple subset names here - either a single string, comma-separated, or a list of strings. Do we need both methods?

That said, the API change I propose above will resolve much of this confusion after the change, so I'm highlighting this now but expect it to go away after your re-write.